include/seqan3/range/view/single_pass_input.hpp

```diff
@@ -0,0 +1,402 @@
+     * \{
+     */
+    //!\brief Compares iterator with sentinel.
+    template <typename sentinel_t>
+    bool operator==(single_pass_input_sentinel<sentinel_t> const & rhs) const noexcept
+    //\cond
+        requires sentinel_concept<std::decay_t<sentinel_t>, std::decay_t<base_iterator_t>>
+    //\endcond
+    {
+        return cached() == rhs.sentinel;
+    }
+
+    //!\brief Compares iterator with sentinel.
+    template <typename sentinel_t>
+    bool operator!=(single_pass_input_sentinel<sentinel_t> const & rhs) const noexcept
+    //\cond
+        requires sentinel_concept<std::decay_t<sentinel_t>, std::decay_t<base_iterator_t>>
+    //\endcond
+    {
+        return !(*this == rhs);
+    }
+    //!\}
+
+protected:
+//!\privatesection
+    //!\brief Gives access to the cached iterator.
+    base_iterator_t & cached() const noexcept
+    {
+        return view_ptr->state_ptr->cached_urng_iter;
+    }
+};
+}  // seqan3::detail
+
+//-----------------------------------------------------------------------------
+// View shortcut for functor.
+//-----------------------------------------------------------------------------
+
+namespace seqan3::view
+{
+/*!\brief A view adapter to add single pass input behavior to the underlying range.
+ * \tparam urng_t The type of the range being processed. See below for requirements.
+ * \param urange The range being processed.
+ * \returns A range with single pass input behavior. See below for the properties of the returned range.
+ * \ingroup view
+ *
+ * ### View properties
+ *
+ * |                     | `urng_t` (underlying range type)      | `rrng_t` (returned range type)                                               |
+ * |---------------------|---------------------------------------|------------------------------------------------------------------------------|
+ * | range               | seqan3::input_range_concept           | seqan3::view_concept + seqan3::input_range_concept + single_pass semantics   |
+ * | `range_reference_t` | `reference_t<urng_t>`                 | `std::add_const_t<reference_t<urng_t>>`                                      |
+ *
+ * * The input properties are **requirements** on the underlying range type.
+ * * The return properties are **guarantees** given on the range return type.
+ * * for more details, see \ref view.
+ *
+ * ### Example
+ *
+ * ```cpp
+ * std::string str{"hello"};
+ * auto v = str | view::single_pass_input;
+ * std::cout << *++v.begin() << std::endl;  // prints 'e'
+ * std::cout << *++v.begin() << std::endl;  // prints 'l'
+ * std::cout << *++v.begin() << std::endl;  // prints 'l'
+ * std::cout << *++v.begin() << std::endl;  // prints 'o'
+ * ```
+ * \hideinitializer
+ */
+detail::declare_view_functor_type<detail::single_pass_input_view> const single_pass_input;
+
+} // namespace seqan3::view
+
+namespace ranges
+{
+    template<typename urng_t>
+    struct enable_view<::seqan3::detail::single_pass_input_view<urng_t>> : std::true_type
+    {};
+}
```

Comment:
you don't need this if you inherit from `view_base`. If you actually do, please file a bug with the ranges library, reference this here and mark this snippet for removal once the issue is resolved.

Reply:
I think it is some left over to figure out, why jenkins builds were causing this issue. Which now gone. So I'll remove it.